Pip3

matkrak opened the pull request:
Pierwsza wersja poprawek mających na celu udostępnienie biblioteki przez narzędzie pip.
Proszę przetestować działanie, jeśli wszystko będzie ok trzeba ustalić jak dokładnie powinny wyglądać metadane wpisane przeze mnie "roboczo".

matkrak committed:
Rename README.md to README.txt

README lub README.txt jest potrzebny do poprawnego działania pip

matkrak committed:
Create setup.py

Plik utworzony na potrzeby dodania biblioteki do listy pakietów obsługiwanej przez pip

grzanka commented:
Chcialbym robic instalacje bezposrednio z repozytorium. Zeby bylo prosciej testuje z Pana repozytorium i cos nie dziala.

```
Leszeks-MacBook-Air:~ grzanka$ sudo pip3 install https://github.com/matkrak/beprof.git
Collecting https://github.com/matkrak/beprof.git
  Downloading https://github.com/matkrak/beprof.git
     - 36kB 278kB/s
  Cannot unpack file /private/tmp/pip-qr6tv1ep-unpack/beprof.git (downloaded from /tmp/pip-t722btz7-build, content-type: text/html; charset=utf-8); cannot detect archive format
Cannot determine archive format of /tmp/pip-t722btz7-build
```

matkrak commented:
co do pip3: mam umożliwić instalację w ten sposób który Pan zaproponował (z repozytorium)? 
beprof jest dodany do zbioru pakietów PyPi i można go zainstalować wpisując
~$ pip install beprof

grzanka commented:
I propose to switch to English everywhere. 
Concerning pip - it should work with GIT now. Remove this package from PyPi - its not mature enough to be in such repository.

grzanka commented on setup.py:
```diff
@@ -0,0 +1,20 @@
+setup(
+    name = 'beprof',
+    packages = ['beprof'],
+    version = '1.0',
```
Make it version 0.1.0. Read more about versioning of the software, i.e. here: http://semver.org, especially fragments related to 

> "How should I deal with revisions in the 0.y.z initial development phase?"
> "How do I know when to release 1.0.0?"

matkrak committed:
Update setup.py

Changes mentioned in comments before

matkrak committed:
Create __init__.py

Testing instalation from git

matkrak committed:
Rename curve.py to beprof/curve.py

changing directory

matkrak committed:
Rename profile.py to beprof/profile.py

changed directory

matkrak commented:
I am now able to install beprof from repo. Command needed (Ubuntu) is 
`~$ pip3 install git+https://github.com/matkrak/beprof.git@pip3`

where "pip3" after @ sign refers to the name of the branch in my repo.

After instalation I am able to import module beprof to my ipython, but there are errors when i try 
`from beprof import profile` and things like this. Work in progress...

matkrak commented:
Not sure if I am doing it the right way, but probaly it closes #1

matkrak committed:
Update and rename README.txt to README.md

Current version probably doesn't need .txt extension to be installed directly from GIT repo,
so i changed it to .md for better performance

matkrak committed:
Update profile.py

Testing import issues

matkrak commented:
Works as it is supposed to be. Test it and give me feedback

grzanka commented on README.md:
```diff
@@ -1,2 +1,22 @@
+stable version is ready it will be pushed to PyPi repo.<br><br>
+
+For now, installation can be done from this GIT repository, using <br>
+`~$ pip3 install git+https://github.com/matkrak/beprof.git@pip3` <br>
```
It works with your repository (tested it). You can put there link to original repo now.

matkrak committed:
Update README.md

Changed link to orginal repo, however it is not going to work untill file structure is changed so that it suits the one in my pip3 branch.

matkrak committed:
Update README.md

Some examples of using beprof added.

grzanka commented on README.md:
````diff
@@ -1,2 +1,59 @@
+>>> print(p)
+```
+
+A few examples of data you can work with using beprof can be downloaded from<br>
````
I would prefer to get the data using urllib directly from the web.
Urllib + genfromtxt (numpy) could solve reading data in 2-3 lines of the code.
Find some interesting dataset online, representing a real profile of something (rising and falling).

grzanka commented on README.md:
````diff
@@ -1,2 +1,59 @@
+```
+>>> import beprof
+>>> from beprof import curve  #imports curve module
+>>> from beprof import profile  #imports profile module
````
When you add the ">>>" at the beginning of the line I get angry when trying to copy-paste. I select block of 3 lines, paste it into python interpreter and get this:

```
Leszeks-MacBook-Air:rbe-o-meter grzanka$ python3
Python 3.5.1 (v3.5.1:37a07cee5969, Dec  5 2015, 21:12:44) 
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> >>> import beprof
  File "<stdin>", line 1
    >>> import beprof
     ^
SyntaxError: invalid syntax
>>> >>> from beprof import curve  #imports curve module
  File "<stdin>", line 1
    >>> from beprof import curve  #imports curve module
     ^
SyntaxError: invalid syntax
>>> >>> from beprof import profile  #imports profile module
  File "<stdin>", line 1
    >>> from beprof import profile  #imports profile module
     ^
SyntaxError: invalid syntax
```

Its annoying to realize that you need to copy line-by-line.

matkrak replied:
I wanted to stress that I am using ipython command line, but now I see.

grzanka commented on README.md:
```diff
@@ -1,2 +1,59 @@
+beprof is based on nparray class from numpy, and it provides <br>
+numerous tools for different computations and data analysis.
+
+# installation
```
Mention that it works only with Python3

grzanka commented on setup.py:
```diff
@@ -0,0 +1,20 @@
+
+from distutils.core import setup
+
+setup(
```
Check what happens if user doesn't have numpy installed.
Its bad idea to instally numpy/scipy using pip, especially on Windows.
Investigate how to put dependency on numpy in setup.py.
It seems there is some discussion about it on stackoverflow: http://stackoverflow.com/questions/25753276/install-numpy-pandas-as-dependency-in-setup-py

matkrak committed:
Update README.md

Removed bad >>> thing

matkrak committed:
Update README.md

And added a line about Python version

matkrak committed:
Update setup.py

Testing install_requires

matkrak committed:
Update setup.py

Testing install_requires ...

matkrak commented:
install_requires field in setup.py works for me. This problem is solved. Proceeding to urllib now...

matkrak committed:
Update setup.py

Forgot scipy

grzanka commented on setup.py:
```diff
@@ -0,0 +1,21 @@
+        "Operating System :: OS Independent",
+        "Topic :: Scientific/Engineering :: Information Analysis",
+    ],
+    install_requires = ['numpy', 'scipy']
```
Is it possible to provide minimum version no of numpy and scipy ?

matkrak replied:
sure right now working on scipy.
By the way which minimum version is required (or I should put up-to-date version)

grzanka replied:
Put up-to-date version (check if you have the same version locally when testing). Later we need to investigate which older versions to support - please create issue for that.

matkrak replied:
OK. I will make separate issue as I handle this scipy thing

matkrak committed:
Update setup.py

testing again

matkrak commented:
Installing scipy via setuptools is very tricky. Many peope advice to write a script in which scipy and numpy are installed via pip3 before installing the main module. I will add scipy to this list once again, try 
pip3 uninstall scipy

and then
pip3 install git+https://github.com/matkrak/beprof.git@pip3

for me it doesn't work and what is more errors refers to libgfortran and openblas which I used in the past for another project. I am trying to find solution but I don't even know what is wrong for now

matkrak committed:
Update setup.py

Just for testing, try installing package after uninstalling scipy and look what happens

grzanka commented:
First and the most important thing - work on clean system. Do you have one ?
A virtualbox machine with minimal installation might be an option (snapshots might be useful to revert system to clean state after playing with install/deinstall of libraries).

I do not want to ask user to install scipy and numpy via pip. Scipy and numpy should be installed from packages (especially Windows).

Ideally pip3 should warn user that numpy and scipy is missing, but not necessary installing them with pip3.

You can also create a separate issue for that numpy-scipy-dependency problem and revert setup.py to previous version. It looks that its not so easy.

matkrak committed:
Update setup.py

Added minimal version of install_requires

matkrak commented:
Does it work for your clean system?
I will install clean Ubuntu tomorrow, I have some things that needs to be done today and I need my system working.
So for now I should leave it as it is and provide some extra information in README ?
or I should remove install_requires field at all?

grzanka commented:
Installing clean Ubuntu inside Virtualbox will not make your system unstable and doesn't require disk formatting.

You can leave it as it is.

I can close this issue and merge PR when new issues are created for tasks which are not finished:
-  examples using urllib and numpy file reading
- careful testing (Windows, Linux) of pip3 behaviour when scipy and numpy is missing or too old

matkrak commented:
2 new issues are created which closes #1 and closes #2 for now.
